src/main/java/com/org/jenkins/custom/jenkins/distribution/service/generators/WarGenerator.java

```diff
@@ -7,25 +7,30 @@
 import java.io.File;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.logging.Logger;
 
 public class WarGenerator {
 
+    private final static Logger LOGGER = Logger.getLogger(WarGenerator.class.getName());
     private static Util util = new Util();
-
-    private static final String PACKAGER_CONFIG_YAML = "packager-config.yml";
     private static final String TEMP_PREFIX = "CDS";
 
-    public static void generateWAR(String versionName) {
+    public static void generateWAR(String versionName, String configuration) {
+        LOGGER.info("Generating War File");
         final Config cfg;
         Path tempDirWithPrefix = null;
         try {
             tempDirWithPrefix = Files.createTempDirectory(TEMP_PREFIX);
-            final File configPath = util.getFileFromResources(PACKAGER_CONFIG_YAML);
-            cfg = Config.loadConfig(configPath);
+            File packagerConfigFile = File.createTempFile("packager-config", ".yml");
+            byte[] buf = configuration.getBytes();
+            Files.write(packagerConfigFile.toPath(), buf);
+            cfg = Config.loadConfig(packagerConfigFile);
             cfg.buildSettings.setTmpDir(tempDirWithPrefix.toFile());
             cfg.buildSettings.setVersion(versionName);
             cfg.buildSettings.setInstallArtifacts(true);
             new Builder(cfg).build();
+            LOGGER.info("Cleaning up temporary directory");
+            packagerConfigFile.deleteOnExit();
         } catch (Exception e) {
             e.printStackTrace();
```

Comment:
I see that the exception is absorbed without consequence to the application. What happens to the application if this fails? Does it continue as if nothing happened?

Reply:
Changed it to throw the exception so that it can be handled later.